x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ExplainLifecycleRequest.java

```diff
@@ -24,26 +31,58 @@
  * Multiple indices may be queried in the same request using the
  * {@link #indices(String...)} method
  */
-public class ExplainLifecycleRequest extends ClusterInfoRequest<ExplainLifecycleRequest> {
+public class ExplainLifecycleRequest extends LocalClusterStateRequest implements IndicesRequest.Replaceable {
```

Comment:
The `ClusterInfo` classes feel a bit off; the only value they add are specifying `indices` and `indicesOptions`, but loads of other actions specify those manually. Therefore, we might as well get rid of the `ClusterInfo` abstraction and let every class define by themselves what fields they have.